src/Command/Setup.php

```diff
@@ -57,18 +62,53 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $io->setDecorated(true);
         $io->comment(sprintf('Composing your <info>%s</info> environment', (string) $envType));
 
-        $vars = new AppVendor(
+        $appVendor = new AppVendor(
             $io->ask('Vendor name', null, [AppVendor::class, 'validate']),
             $io->ask('App name', null, [AppVendor::class, 'validate'])
         );
-        $env = EnvFactory::createEnv($envType);
 
-        foreach (Dumper::dump($env, $vars, $cwd) as $dumpTarget) {
+        $envMetadata = MetadataParser::parse($envType);
+        $postProvisionTask = new MakeTask('install', $envMetadata->get('script.post_provision'));
+        $env = EnvFactory::createEnv($envType, $appVendor, $postProvisionTask, $this->setupDependencies($io, $envMetadata));
+
+        foreach (Dumper::dump($env, $appVendor, $cwd) as $dumpTarget) {
             $io->writeln(sprintf('- %s', str_replace($cwd.'/', '', $dumpTarget)));
         }
 
         $io->success('Environment successfully configured');
 
         return 0;
     }
+
+    private function setupDependencies(SymfonyStyle $io, MetadataBag $metadata)
+    {
+        foreach ($metadata->get('packages') as $name => $settings) {
+            $defaultEnabled = $settings['enabled'] ?: false;
+            $defaultVersion = isset($settings['default']) ? $settings['default'] : null;
+            $enabled = $settings['required'] ?: $io->confirm(sprintf('Install %s?', $name), $defaultEnabled);
+
+            if (null === $defaultVersion) {
+                yield new Dependency($name, $enabled);
```

Comment:
You're right that there is no gain in term of memory at using a generator here. However there is no cost and I find nicer to deal with an Iterator rather than an array here :)